### PR TITLE
DR-1799: TIM name encoding

### PR DESCRIPTION
**Pull request by pshapiro4broad**
Implementation of proposed TIM name mapping from https://docs.google.com/document/d/186mRxKcCmYLh20mj15_ZI2VCtoVQgomCKNH1zU49z_s/edit#

Even though this isn't currently used, we have tickets in the backlog that depend on this code, so I'm changing the PR from Draft to merge it up now.

### CODE REVIEW

File: src/main/java/bio/terra/app/utils/TimUtils.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
+package bio.terra.app.utils;
+
+/**
+ * Utility methods for working with the Terra Interoperability Model (TIM)
+ */
+public final class TimUtils {
+    private TimUtils() { }
+
+    private static final String CAPITOL = "a__";
```
Comment (fboulnois) [inline src/main/java/bio/terra/app/utils/TimUtils.java]: nit: 
```suggestion
    private static final String CAPITAL = "a__";
```

File: docs/jade-getting-started.md [hunk 2]
```diff
@@ -342,8 +342,6 @@ jade-data-repo/application.properties.  You can convert any application.property
 variable by switching to upper case and every "." to "_".
 
 ```
-export DATAREPO_USEREMAIL={your dev gmail account}
-
```
Comment (fboulnois) [inline docs/jade-getting-started.md:346]: This change is unrelated, correct?

File: src/main/java/bio/terra/app/utils/TimUtils.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            t = t.substring(0, index) + Character.toUpperCase(t.charAt(index + len)) + t.substring(index + len + 1);
+        }
+        return t;
+    }
+
+    /**
+     * Give a string, return true if it's likely that this was encoded and requires decoding.
+     *
+     * @param s the string
+     * @return true if it's likley to be encoded
```
Comment (fboulnois) [inline src/main/java/bio/terra/app/utils/TimUtils.java]: nit:
```suggestion
     * @return true if it's likely to be encoded
```

**Review by fboulnois — COMMENTED** [review]
A few minor suggestions, otherwise I like the approach! Since `shouldDecode` is probabilistic, I wonder if we should add a unique prefix like `tim__` in addition / instead so we would only need to check the first 5 characters.

File: src/main/java/bio/terra/app/utils/TimUtils.java (hunk 1 above)
Comment (pshapiro4broad) [reply src/main/java/bio/terra/app/utils/TimUtils.java]: 🤦

File: docs/jade-getting-started.md (hunk 2 above)
Comment (pshapiro4broad) [reply docs/jade-getting-started.md:346]: Yes, I happened to commit it, it's a change I had lying around from before. I can remove it, although it should be updated at some point.

**Comment by pshapiro4broad** [comment]
> A few minor suggestions, otherwise I like the approach! Since shouldDecode is probabilistic, I wonder if we should add a unique prefix like tim__ in addition / instead so we would only need to check the first 5 characters.

I thought about that too, having a special prefix for the encoded name. Now that I think about it, checking for a prefix of `a__terraa__corec` should work, as I believe all TIM data model elements are scoped inside `TerraCore`.

**Review by fboulnois — APPROVED** [review]
latest changes lgtm

**Review by fboulnois — APPROVED** [review]
lgtm

**Review by tcjiang9 — APPROVED** [review]
